documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -233,5 +267,94 @@ data transformation pipelines that also can be used with parallel range algorith
     return 0;
   }
 
+.. _use-iterators:
+
+Use |onedpl_short| Iterators
+----------------------------
+
+|onedpl_short| provides a set of `iterators <iterators-details>`_ that can be used to pass data to algorithms, in
+combination with the iterators described above and with the return of ``oneapi::dpl::begin()`` and
+``oneapi::dpl::end()``. To pass data to an algorithm with a device execution policy with minimum data movement, use
+iterators that are both `SYCL device-copyable`_ and `indirectly device accessible <indirectly-device-accessible>`_.
+These properties of the |onedpl_short| iterators are described in `this table <iterator-properties-table>`_.
+
+Use Custom Iterators
+--------------------
+If the provided iterators are not sufficient for your needs, you can create your own iterators that can be used as input
+to |onedpl_short| algorithms. To pass data efficiently to an algorithm with a device execution policy, the custom
+iterator must be SYCL device-copyable and indirectly device accessible. If a custom iterator is *not* defined to be
+indirectly device accessible, the algorithm will create a temporary SYCL buffer to copy the data from the iterator to
+the device. This may lead to performance degradation, and also requires that the data be accessible on the host to be
+copied into the temporary SYCL buffer.
```

Comment:
```suggestion
to |onedpl_short| algorithms. The custom iterator must be SYCL device-copyable.
If it references data accessible on a device, it also must be indirectly device accessible.

.. note::

  If a custom iterator is *not* defined to be indirectly device accessible, 
  the algorithm will create a temporary SYCL buffer to copy the data from the iterator to the device. 
  This may lead to performance degradation, and also requires that
  the data be accessible on the host to be copied into the temporary SYCL buffer.
```